File: app/src/main/java/com/androideas/quakesnz/app/model/Geometry.java
package com.androideas.quakesnz.app.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Geometry implements Parcelable {

    public static final Parcelable.Creator<Geometry> CREATOR = new Creator<Geometry>() {

        @Override
        public Geometry[] newArray(int size) {
            return new Geometry[size];
        }

        @Override
        public Geometry createFromParcel(Parcel source) {
            return new Geometry(source);
        }
    };

    private float[] coordinates;

    public Geometry(Parcel source) {
        coordinates = new float[source.readInt()];
        source.readFloatArray(coordinates);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public float[] getCoordinates() {
        return coordinates;
    }

    public void setCoordinates(float[] coordinates) {
        this.coordinates = coordinates;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(coordinates.length);
        dest.writeFloatArray(coordinates);
    }

}

File: core/src/main/java/nz/co/codebros/quakesnz/core/model/Geometry.kt
package nz.co.codebros.quakesnz.core.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created by leandro on 3/07/17.
 */

data class Geometry(
        @SerializedName("type")
        val type :String,

        @SerializedName("coordinates")
        val coordinates: Coordinates
) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readParcelable(Coordinates::class.java.classLoader)
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(type)
        parcel.writeParcelable(coordinates, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Geometry> {
        override fun createFromParcel(parcel: Parcel): Geometry {
            return Geometry(parcel)
        }

        override fun newArray(size: Int): Array<Geometry?> {
            return arrayOfNulls(size)
        }
    }
}